imap/index.c

```diff
@@ -3110,11 +3111,13 @@ index_copy(struct index_state *state,
 done:
     free(copyargs.records);
     mailbox_close(&destmailbox);
-    for (i = 0; i < msgrecs->count; i++) {
-        msgrecord_t *mr = ptrarray_nth(msgrecs, i);
-        msgrecord_unref(&mr);
+    if (msgrecs) {
```

Comment:
`msgrecs` cannot be NULL here.

Wouldn’t be sufficient to move the `ptrarray_t *msgrecs = ptrarray_new();` from line 2992 to line 3008?

Reply:
@dilyanpalauzov the `goto done` jump in line 3023 could jump over the ptrarray_new in line 3053.

Reply:
`msgrecs` can indeed be NULL here, and one of the Cassandane tests even crashes on it without the conditional

These might be tiny changes, but I did actually spend time thinking about them, and I'd appreciate if you respected that.